woocommerce/woocommerce-ios
# [What's New Component] Fetch and Store Feature Announcements on App Launch

Part of: #4862

# Description 📜

This PR introduces the main mechanism of fetching new announcements to warn users about the newest released features of the current app version.

For now, when user opens the app, an action for fetching the announcements is being dispatched but doing nothing with the response (as this will be implemented on #4863).

More information about the endpoint that is being used can be found [here](https://fieldguide.automattic.com/mobile-feature-announcements/mobile-feature-announcement-endpoint/)

# Main Changes ⚙️
- Added a new _Store_ named `AnnouncementsStore`, responsible for fetching _Announcements_ and persisting its _Features_  on disk (.plist file). 
- Dispatch a new action (`AnnouncementsAction.synchronizeFeatures`) on app coordination start.
- Inject _AnnouncementServiceRemote_ into Yosemite (as we are going to user a pre-existing this WordPressKit service remote to fetch the latest announcements). For this, a new protocol to create the boundary was created with the name `AnnouncementsRemoteProtocol`.
- The Features are being mapped into _WooCommerceFeature_, a model that will be displayed on the Whats New Component screen.

# Screenshots 📱
![image](https://user-images.githubusercontent.com/997521/130534138-a04739b6-0915-4912-b22f-7f8f326ac8e4.png)


# Testing 🧑‍🔬🧪

1. Launch the app
2. Go to Settings (top-right corner of the NavigationBar)
3. Tap on `Launch Wormholy Debug`
4. Filter for "announcements"
5. There should be a request for the announcements endpoint

### If you want to test it with real data, please do the following change:
<img width="495" alt="image" src="https://user-images.githubusercontent.com/997521/131144031-136d2892-cb05-495a-8532-dbc924d8ee02.png">


⚠️ As of today, there are no versions with Announcements, so the caching mechanism cannot be tested (unless you use a network app to stub the network responses, like [MockingBird](https://github.com/Farfetch/mockingbird) or [Charles proxy](https://www.charlesproxy.com))

## Update release notes:
